Modules/_sre/sre.c

```diff
@@ -1508,6 +1508,9 @@ _sre_compile_impl(PyObject *module, PyObject *pattern, int flags,
     for (i = 0; i < n; i++) {
         PyObject *o = PyList_GET_ITEM(code, i);
         unsigned long value = PyLong_AsUnsignedLong(o);
+        if (value == (unsigned long)-1 && PyErr_Occurred()) {
+            break;
```

Comment:
IMO, we should just bail here. That means we can remove the `PyErr_Occurred` check in 1523. What do you think?

Reply:
I would prefer to keep it as-is, the main reason is that the change is minimal now. I don't want to touch other stuff :)

Reply:
I'm fine with that :)